src/ddscxx/src/org/eclipse/cyclonedds/topic/qos/TopicQosDelegate.cpp

```diff
@@ -198,6 +207,7 @@ TopicQosDelegate::named_qos(const struct _DDS_NamedTopicQos &qos)
     priority_    .delegate().v_policy((v_transportPolicy&)       (q->transport_priority));
     lifespan_    .delegate().v_policy((v_lifespanPolicy&)        (q->lifespan)          );
     ownership_   .delegate().v_policy((v_ownershipPolicy&)       (q->ownership)         );
+    datarepresentation_.delegate().v_policy((v_dataRepresentationPolicy&)(q->datarepresentation));
```

Comment:
Please also encapsulate this line with the conditional OMG_DDS_EXTENSIBLE_AND_DYNAMIC_TOPIC_TYPE_SUPPORT macro. (BTW: I see that in this particular case the embedding OMG_DDS_OWNERSHIP_SUPPORT macro was also missing for the "ownership" policy).

Reply:
Will do.